pxtlib/auth.ts

```diff
@@ -305,17 +312,19 @@ namespace pxt.auth {
             return result.success;
         }
 
+        private debounceResult: Promise<boolean>;
+
         private prefPatchOps: ts.pxtc.jsonPatch.PatchOperation[] = [];
 
-        public async patchUserPreferencesAsync(ops: ts.pxtc.jsonPatch.PatchOperation | ts.pxtc.jsonPatch.PatchOperation[], immediate = false) {
+        public async patchUserPreferencesAsync(ops: ts.pxtc.jsonPatch.PatchOperation | ts.pxtc.jsonPatch.PatchOperation[], immediate = false): Promise<SetPrefResult> {
             ops = Array.isArray(ops) ? ops : [ops];
             ops = ops.filter(op => !!op);
-            if (!ops.length) { return; }
             const curPref = await this.userPreferencesAsync();
+            if (!ops.length) { return {success: true, res: curPref} }
             ts.pxtc.jsonPatch.patchInPlace(curPref, ops);
             await this.setUserPreferencesAsync(curPref);
             // If we're not logged in, non-persistent local state is all we'll use
-            if (!await this.loggedInAsync()) { return; }
+            if (!await this.loggedInAsync()) { return {success: false, res: undefined}; }
```

Comment:
I feel like this should return `{ success: true, res: currPref }`. It isn't an error condition to be signed out.

Reply:
Nit/Suggestion: add whitespace after `{` and before `}`. We could use a lint rule for this.

Reply:
hmm but if you're signed out then the choice won't be saved? I guess at least in email case you can't even change it

Reply:
If you're signed out, right now the profile page will be inaccessible. Maybe that will change one day.